### PR TITLE
Rebalances Hivemind Special Options

### PR DESCRIPTION
**Pull request by SingingSpock**
## About The Pull Request
<details>
<summary>
	Spreading through burrows now no longer refuses to spread if the burrow is in maintenance, and gibbing players now gives a warning and 90 seconds first
</summary>
<hr>

Firstly, the spreading check. When "Spread through burrows" is enabled, the wires will now spread even if the origin burrow is in maintenance. This means that it actually does spread, since most burrows are in maintenance. Testing has verified that the hivemind now does properly spread in hard-to-predict ways.

Secondly, the gibbing check. Previously, it would happen almost instantly if the option was enabled, which isn't very fun player-side. If something *is* going to RR them, there should be more ways to prevent it. Therefore it now gives a message about wires wreathing around them once the hivemind starts the process of assimilating them, and if they're not on the space 90 seconds later it cancels the assimilation.	
<hr>
</details>

## Changelog
:cl:
tweak: When "Spread through burrows" is enabled for the hivemind, it no longer refuses to spread through maintenance burrows.
tweak: The  "Prevent gibbing players" has been replaced with a timer that can be set for how long a body takes to be assimilated and removed. It defaults to 0, which completely disables assimilation of humans
fix: Hivemind turrets will now actually shoot people
/:cl:

### CODE REVIEW

File: code/modules/hivemind/wires.dm [hunk 1]
```diff
@@ -389,6 +389,11 @@
 			if(is_neotheology_disciple(L))
 				return
 
+			to_world("Wires begin to wreathe around [L], starting the process of converting them into part of the hivemind.") //We tell people to get them off the wires
+			sleep(90 SECONDS) //We give them 90 seconds to be pulled off the wires
```
Comment (Trilbyspaceclone) [inline code/modules/hivemind/wires.dm:393]: Sleep is really bad to use, as it stops here for 90 seconds and returns the proc constantly asking `is 90nseconds past?`

Comment (SingingSpock) [reply code/modules/hivemind/wires.dm:393]: What's a better alternative to use?

File: code/modules/hivemind/wires.dm [hunk 2]
```diff
@@ -381,20 +381,20 @@
...
-				L.drop_from_inventory(W)
-			var/M = pick(/mob/living/simple_animal/hostile/hivemind/himan, /mob/living/simple_animal/hostile/hivemind/phaser)
-			new M(loc)
+			visible_message("Wires begin to wreathe around [L], starting the process of converting them into part of the hivemind.") //We tell people to get them off the wires
+			addtimer(CALLBACK(src, .proc/assimilate_human, L), timer)
+			return
+
 		//robot corpses
-		else if(issilicon(subject))
+		else if(issilicon(subject)) //If you're a borg... sucks to suck? I don't feel like reworking this, you're too mechanical to prevent hivemind taking over you
```
Comment (Trilbyspaceclone) [inline code/modules/hivemind/wires.dm:397]: then fpbs should get instantly converted to >:T

File: code/modules/hivemind/wires.dm [hunk 3]
```diff
@@ -403,6 +403,15 @@
 
 		qdel(subject)
 
+/obj/effect/plant/hivemind/proc/assimilate_human(var/mob/living/L)
+	if(!locate(/obj/effect/plant/hivemind) in L.loc) //If we don't see any wires after the alotted time, we let them go
```
Comment (Dimasw99) [inline code/modules/hivemind/wires.dm:407]:
```suggestion
	if(!locate(/obj/effect/plant/hivemind) in L.loc || !(L.stat == DEAD)) //If we don't see any wires after the alotted time, we let them go
```

Comment (Dimasw99) [reply code/modules/hivemind/wires.dm:407]: A sanity check incase we were revived and moved back to the location

File: code/modules/hivemind/wires.dm [hunk 4]
```diff
@@ -381,20 +381,20 @@
...
+
+			if(is_neotheology_disciple(L)) //If our target has a cruciform, we don't touch it
 				return
 
-			for(var/obj/item/W in L)
-				L.drop_from_inventory(W)
-			var/M = pick(/mob/living/simple_animal/hostile/hivemind/himan, /mob/living/simple_animal/hostile/hivemind/phaser)
-			new M(loc)
+			visible_message("Wires begin to wreathe around [L], starting the process of converting them into part of the hivemind.") //We tell people to get them off the wires
+			addtimer(CALLBACK(src, .proc/assimilate_human, L), timer)
```
Comment (Dimasw99) [inline code/modules/hivemind/wires.dm:393]:
```suggestion
			timer_id = addtimer(CALLBACK(src, .proc/assimilate_human, L), timer, TIMER_STOPPABLE)
```

Comment (Dimasw99) [reply code/modules/hivemind/wires.dm:393]: Our weeds are going to be cut and destroyed. Github is being annoying but you also need to define a basic var/timer_id at the top and deltimer(timer_id) for both Destroy() and die_off(). Should only be:

```
/obj/effect/plant/hivemind/die_off()
	if(timer_id)
		deltimer(timer_id)
	return ..()
```

**Pull request merged**